### DISCUSSION TITLE
French category not showing up

### DISCUSSION THREAD

**Discussion by willinrecovery** [code4recovery/12-step-meeting-list#1395] [Q&A] [answered]
We discovered the plugin seems to support multiple languages tonight.  Awesome! We are in Canada so this is great!

We noticed one odd thing, however. When a French visitor goes to look for a French meeting there is no option to search for French meetings. It turns out there is not option to search for French meetings for an English speaker too, I just hadn't noticed it.

So I went into the Meetings dashboard to see if the meetings were marked as French. They are not, although I could swear there were previously options for a handful of languages. In fact, all of the English meetings are successfully categorized as English even though we are not presented with the option to do so — so it must have been there at one point.

Have I done something to reduce the options available?

https://staging.debtorsanonymous.ca/

Thank you so much for your service!

**Comment by SIA-WEBSUP** [comment]
There must be at least one meeting typed as French before you'll see the option to filter on this type. This is true for every 'type' of meeting.

In the admin screen, do you see a 'View all' button near the bottom of the section that lists the 'Types'.
![image](https://github.com/code4recovery/12-step-meeting-list/assets/63924390/7dab5df2-391f-4ea7-9b68-6fcda68e23ed)


When I hit this button on my test site I'm able to see all supported languages including French:
![image](https://github.com/code4recovery/12-step-meeting-list/assets/63924390/d1c6b95a-2743-4694-ae52-8be562900777)

**Comment by willinrecovery** [comment]
Yes, I have that button, but I don't see the languages.

Maybe it has something to do with us using the Debtors Anonymous categories? I don't understand how I managed to categorize some of them as English.

<img width="906" alt="Monosnap CECDA Staging (admin) 2024-03-21 12-05-52" src="https://github.com/code4recovery/12-step-meeting-list/assets/159723102/32d33e9a-b87b-448c-baeb-d123b2e83093">

**Reply by SIA-WEBSUP** [reply]
I think you may be right about having only limited languages to choose from when the Program is set to DA. I'm not 100% why this is the case, but I will investigate further.

It may be that all meetings default to 'English' since it wouldn't make sense for a meeting to have no language type set.

Just out of curiosity, what happens if you change the Program to AA, set the languages you want, and then change the Program back to DA?

**Reply by willinrecovery** [reply]
Setting the program to AA temporarily, checking the right language box and switching it back to DA does work as a workaround between now and when the problem is resolved.  Notably, the English meetings are not tagged as English so it seems to be a default. The French filter also appears as would be expected.

I will do this and categorize all the meetings again when I have time this week / weekend.

**Reply by willinrecovery** [reply, marked as answer]
Okay, this worked. But I decided it was problematic because it could be confusing for future web admins.

So I tried to create my own French and English meeting types specific to Debtors Anonymous, but I discovered this was not necessary. 

When I added the following code to my functions.php in my child theme, it made the English and French settings appear even when set to Debtors Anonymous (sorry I can't figure out how to paragraph this code properly).

`if (function_exists('tsml_custom_types')) {
    tsml_custom_types(array(
        'EN' => 'English',
        'FR' => 'French',
    ));
}`

**Reply by SIA-WEBSUP** [reply]
FYI, there is an open Issue that I believe will fix this bug. But I don't know how soon it will  be rolled out. 
 - https://github.com/code4recovery/12-step-meeting-list/issues/1252

Terrific that you found a satisfactory workaround!
